sys/net/link_layer/ieee802154/submac.c

```diff
@@ -337,6 +337,43 @@ int ieee802154_submac_init(ieee802154_submac_t *submac, const network_uint16_t *
         submac->channel_page = CONFIG_IEEE802154_DEFAULT_SUBGHZ_PAGE;
     }
 
+    /* Get supported PHY modes */
+    int phy_modes = 0;
+    if (ieee802154_radio_has_phy_bpsk(dev)) {
```

Comment:
if we change the `get_caps` to a flag based solution, we could allocate the bitmask beforehand an simply do e.g:

```c
phy_modes = dev->caps & IEEE802154_PHY_MODES_MASK;
```

Reply:
but I would keep these ieee802154_radio_has_phy_xxx, since those can be handy anyway :).

Reply:
>if we change the get_caps to a flag based solution, we could allocate the bitmask beforehand an simply do e.g:

Should it be provided as a separate PR then? Or could I modify it directly here on this PR.

Reply:
I think in any case in a different PR. Since the Radio HAL [is still under discussion](https://github.com/RIOT-OS/RIOT/pull/13943), I think it's worth to comment it in the RDM.